src/main.cpp

```diff
@@ -5185,11 +5211,16 @@ static bool AcceptBlock(const CBlock& block, CValidationState& state, const CCha
             blockPos = *dbp;
         if (!FindBlockPos(state, blockPos, nBlockSize+8, nHeight, block.GetBlockTime(), dbp != NULL))
             return error("AcceptBlock(): FindBlockPos failed");
-        if (dbp == NULL)
-            if (!WriteBlockToDisk(block, blockPos, chainparams.MessageStart()))
+
+        if (dbp == NULL) {
+            if (!WriteBlockToDisk(block, blockPos, chainparams.MessageStart())) {
                 AbortNode(state, "Failed to write block");
-        if (!ReceivedBlockTransactions(block, state, chainparams, pindex, blockPos))
+            }
+        }
+        setDirtyBlockIndex.insert(pindex);
```

Comment:
I didn't remember this move; what's the rationale for the move here?

Reply:
`setDirtyBlockIndex` can't be modified within `ReceivedBlockTransactions` because that causes UB in the `Validation.ReceivedBlockTransactions` gtest.